runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -409,6 +428,45 @@ function LanguageTree:_add_injections()
   return query_time
 end
 
+--- Run an asynchronous parse, calling {on_parse} when complete.
+---
+--- @param range boolean|Range|nil
+--- @param on_parse fun(trees: TSTrees, timeout_reached: boolean)
```

Comment:
nit: "expired" is the usual term I think

```suggestion
--- @param on_parse fun(trees: TSTrees, timeout_expired: boolean)
```